File: web/Entity.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
function Entity() {
    this.direction = "UP";
    this.locX = 0;
    this.locY = 0;
    this.isDead = false;
    this.hasPowerUp = false;
    this.health = 1;
    this.canMove = true;
    this.sprite = new Image;
    this.movementSpeed = 1;
    this.bulletSpeed = 2;
    this.move = function (directionToMove) {
        direction = directionToMove;
        if (directionToMove === Direction.up) {
            locY-=movementSpeed;
        }
        if (directionToMove === Direction.down) {
            locY+=movementSpeed;
        }
        if (directionToMove === Direction.left) {
            locX-=movementSpeed;
        }
        if (directionToMove === Direction.right) {
            locX+=movementSpeed;
        }
    };
    this.fire = function(){
        
    };
}
 
Entity.prototype.getInfo = function() {
    return this.color + ' ' + this.type + ' apple';
};

function Direction(){
    this.left = "LEFT";
    this.right = "RIGHT";
    this.up = "UP";
    this.down = "DOWN";
}

File: src/java/Game/Entity.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Game;

/**
 *
 * @author Chris
 */
public class Entity {
    private int locX, locY, health, movementSpeed;
    private Direction direction;
    private boolean isAlive, isInvulnerable;
    
    public Entity(int x, int y, Direction direc, int health){
        direction = direc;
        locX = x;
        locY = y;
        isAlive = true;
        isInvulnerable = true;
        this.health = health;
        movementSpeed = 1;
    }
    
    public void go(Direction direction){
        this.direction = direction;
        switch(direction){
            case LEFT:
                if(locX - movementSpeed > Game.BOARD_WIDTH_MIN)
                    locX-= movementSpeed;
                break;
            case RIGHT:
                if(locX + movementSpeed <= Game.BOARD_WIDTH_MAX)
                    locX+= movementSpeed;
                break;
            case UP:
                if(locY - movementSpeed > Game.BOARD_HEIGHT_MIN)
                    locY-= movementSpeed;
                break;
            case DOWN:
                if(locY + movementSpeed <= Game.BOARD_HEIGHT_MAX)
                    locY+= movementSpeed;
                break;
            default:
                break;
        }
    }
    
    public void hit(int damage){
        health -= damage;
        if(health <= 0)
            this.isAlive = false;
        
    }
    
    public enum Direction{
        LEFT, RIGHT, UP, DOWN
    }
    
    public class Health{
        static final int DEFAULT_HEALTH = 1;
        
    }
}
